lib/maths/CPeriodicityHypothesisTests.cc

```diff
@@ -2369,6 +2369,24 @@ CPeriodicityHypothesisTests::CNestedHypotheses::CBuilder::finishedNested() {
 
 namespace {
 
+//! Apply selection bias for known common periods.
+double applySelectionBias(std::size_t periodBuckets, core_t::TTime bucketLength, double correlation) {
+    core_t::TTime period{static_cast<core_t::TTime>(periodBuckets) * bucketLength};
+    if (std::abs(period - core::constants::YEAR) <= bucketLength) {
```

Comment:
Is it worth adding a comment to say this is being done for years because years vary in length?  Or if that isn't the reason then it's not clear why this is done for year but not more common periods.

Reply:
The reason for this is that we estimate the period, via maximum cyclic autocorrelation, and this estimate can be in error. If the estimated period is near a year it is likely that the period is actually one year. It is advantageous to use a period as close as possible to the true period since this minimises precession.

Other common periods are already handled: daily and weekly are dealt with explicitly when testing for seasonal components. I debated treating average monthly periodicity as special case as well, but we have a variety of predictive calendar features. These are usually more suitable, since the variation from month to month in the monthly period is relatively large.

Reply:
I'll add a comment.

Reply:
See [93cbad3](https://github.com/elastic/ml-cpp/pull/1297/commits/93cbad37cbef56122803d5743d9ef6989eef8fd0).